### ISSUE TITLE
About the Augmented dataset

### ISSUE THREAD

**Issue by liuhengyue** [waxnkw/IETrans-SGG.pytorch#19]
Hi,

This is a really cool work!

I have a few question regarding the provided dataset annotation found in ModelZoo. I checked the one named I-70% + E-100%, and it contains a list 58550 entries. Does it mean these contain the modified groundtruth (GT) based on internal and external transfer? So not all GT are changed, and we should use the original annotations for those are not included in this file? For modified ones, are they the only annotations to be used for training a particular image, or do we concatenate them with the original annotations?

Also, I am still confused on 'possible_rels' even though I looked over other issues. This field is image-specific and as a list of lists of predicate labels, how do I use it? Or it is not used.

Thanks for your time!

**Comment by waxnkw** [comment]
Q1:  Does it mean these contain the modified groundtruth (GT) based on internal and external transfer?
A1: Yes.

Q2: So not all GT are changed, and we should use the original annotations for those are not included in this file?
A2: No, the unchanged annotations are also included in this file. It is merged by the changed ones and the original annotation.

Q3: For modified ones, are they the only annotations to be used for training a particular image, or do we concatenate them with the original annotations?
A3: it is already a merged file.

Q4: confusion about 'possible_rels'.
A4: For external transfer, I add a limitation that only the relation triplet types that appeared in the training set can be added. For example, if there is no (man, drinking, horse) shown in the training set, it will be not added. The "possible_rels" is actually possible relationships between different pairs.
For the final I-70% + E-100% file, it will be not used. It will only be used for generating the E100% file, where E-100% will be merged with I-70% for the final I-70% + E-100% file.

**Comment by liuhengyue** [comment]
Thanks for your answers!

**Comment by bowen-upenn** [comment]
Could you please clarify the format of `possible_rels` if we plan to use `ietrans-em_E.pk_0.7` as a new dataset for training our SGG model? 

The sublists in `possible_rels` are not in triplet format, so how can we link each sublist in `possible_rels` to their corresponding subject and object?  Additionally, how can we determine which integers represent relations or object labels, and which possible predicate has the highest attraction or prediction score? Does `relations` contain groundtruth triplets of format [subject label, relation label, object label] in the original dataset? Thank you!

For example, the first element in the new dataset annotation `ietrans-em_E.pk_0.7` is 
```
{'img_path': 'datasets/vg/VG_100K/99.jpg', 'boxes': array([[224.375,  40.   , 392.5  , 344.375],
       [408.125, 273.125, 636.25 , 447.5  ],
       [546.875, 285.625, 638.125, 476.875],
       [196.875,  28.75 , 236.875, 345.   ],
       [422.5  ,   0.   , 637.5  , 343.125],
       [420.625,  11.875, 500.625, 341.25 ],
       [253.75 , 220.625, 290.   , 303.125],
       [257.5  , 351.875, 288.75 , 425.   ],
       [232.5  , 346.875, 260.   , 425.   ]], dtype=float32), 
'labels': array([ 38, 126, 108,  38,  22,  22, 108,  54,  54]), 
'possible_rels': [[5, 29, 30, 31, 50], [6, 20], [22, 23, 29], [22, 23, 29], [4, 6, 8, 29, 31], [5, 29, 30, 31, 50], [8, 29, 31], [4, 7, 8, 10, 20, 22, 23, 29, 30, 31, 32], [8, 29, 31], [4, 7, 8, 10, 20, 22, 23, 29, 30, 31, 32], [8, 9, 22, 29, 30, 31, 48, 50], [8, 9, 22, 29, 30, 31, 48, 50]], 
'width': 640, 'height': 480, 'relations': array([[2, 1, 6]])}
```
